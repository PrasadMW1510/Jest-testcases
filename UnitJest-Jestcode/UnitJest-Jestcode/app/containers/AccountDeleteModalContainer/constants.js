/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
