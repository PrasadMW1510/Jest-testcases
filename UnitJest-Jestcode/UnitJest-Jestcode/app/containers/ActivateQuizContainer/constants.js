/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
