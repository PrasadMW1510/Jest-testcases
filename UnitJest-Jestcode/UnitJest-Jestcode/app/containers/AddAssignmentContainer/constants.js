/*
 *
 * AddAssignmentContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/AddAssignmentContainer/DEFAULT_ACTION';
export const GET_STUDENTDETAIL_SUCCESS = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENTDETAIL_FAILURE = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_STUDENT_DETAILS = 'app/AddAssignmentContainer/GET_STUDENT_DETAILS';
export const POST_NEW_ASSIGNMENT = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT';
export const POST_NEW_ASSIGNMENT_SUCCESS = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS';
