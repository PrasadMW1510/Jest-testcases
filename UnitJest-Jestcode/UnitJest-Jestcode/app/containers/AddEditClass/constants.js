/*
 *
 * AddEditClass constants
 *
 */

export const SAVE_CLASS_REQUEST = 'AddEditClass/save_class_request';
export const SAVE_CLASS_REQUEST_SUCCESS = 'AddEditClass/save_class_request_success';
export const SAVE_CLASS_REQUEST_FAILURE = 'AddEditClass/save_class_request_failure';
export const SAVE_CLASS_MIA_REQUEST = 'AddEditClass/save_class_mia_request';
export const SAVE_CLASS_MIA_REQUEST_SUCCESS = 'AddEditClass/save_class_mia_request_success';
export const SAVE_CLASS_MIA_REQUEST_FAILURE = 'AddEditClass/save_class_mia_request_failure';

export const INITIALIZE_CLASS_FORM_REQUEST = 'AddEditClass/initialize_class_form_request';
export const INITIALIZE_CLASS_FORM_REQUEST_SUCCESS =
  'AddEditClass/initialize_class_form_request_success';
export const INITIALIZE_CLASS_FORM_REQUEST_FAILURE =
  'AddEditClass/initialize_class_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_TEACHERS = 'meta_data_teachers';
export const META_DATA_STUDENTS = 'meta_data_students';
export const FIELD_SELECTED_STUDENTS = 'rosterStudents';
export const FORM_CLASS_PROFILE = 'form_class_profile';
export const FORM_TITLE_ADD = 'Add a Class';
export const FORM_TITLE_EDIT = 'Edit Class Profile';
