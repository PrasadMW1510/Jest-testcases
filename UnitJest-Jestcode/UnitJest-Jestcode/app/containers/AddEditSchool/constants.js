/*
 *
 * AddEditSchool constants
 *
 */

export const FORM_SCHOOL_PROFILE = 'form_school_profile';
export const FORM_TITLE_ADD = 'Add a School';
export const FORM_TITLE_EDIT = 'Edit School Profile';

// Actions

export const META_DATA_GRADES = 'meta_data_grades';

export const GRADE_LIST_FOR_DISTRICT = 'AddEditSchool/grade_list_for_district';
export const GET_META_DATA_REQUEST = 'AddEditSchool/get_meta_data_request';
export const GET_META_DATA_REQUEST_SUCCESS = 'AddEditSchool/get_meta_data_request_success';
export const GET_META_DATA_REQUEST_FAILURE = 'AddEditSchool/get_meta_data_request_failure';

export const INITIALIZE_SCHOOL_FORM_REQUEST = 'AddEditSchool/initialize_school_form_request';
export const INITIALIZE_SCHOOL_FORM_REQUEST_SUCCESS =
  'AddEditSchool/initialize_school_form_request_success';
export const INITIALIZE_SCHOOL_FORM_REQUEST_FAILURE =
  'AddEditSchool/initialize_school_form_request_failure';

export const SAVE_SCHOOL_REQUEST = 'AddEditSchool/save_school_request';
export const SAVE_SCHOOL_REQUEST_SUCCESS = 'AddEditSchool/save_school_request_success';
export const SAVE_SCHOOL_REQUEST_FAILURE = 'AddEditSchool/save_school_request_failure';

export const SAVE_SCHOOL_MIA_REQUEST = 'AddEditSchool/save_school_mia_request';
