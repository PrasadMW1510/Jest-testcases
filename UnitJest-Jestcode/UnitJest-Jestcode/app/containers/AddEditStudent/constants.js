/*
 *
 * AddEditStudent constants
 *
 */

export const SAVE_STUDENT_REQUEST = 'AddEditStudent/save_student_request';
export const SAVE_STUDENT_REQUEST_SUCCESS = 'AddEditStudent/save_student_request_success';
export const SAVE_STUDENT_REQUEST_FAILURE = 'AddEditStudent/save_student_request_failure';
export const FORM_STUDENT_PROFILE = 'FORM_STUDENT_PROFILE';
export const INITIALIZE_STUDENT_FORM_REQUEST = 'AddEditStudent/initialize_student_form_request';
export const INITIALIZE_STUDENT_FORM_REQUEST_SUCCESS =
  'AddEditStudent/initialize_student_form_request_success';
export const INITIALIZE_STUDENT_FORM_REQUEST_FAILURE =
  'AddEditStudent/initialize_student_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_CLASSES = 'meta_data_classes';
export const META_DATA_GROUPS = 'meta_data_groups';
export const META_DATA_PASSWORD_CONFIG = 'meta_data_password_config';
export const FORM_TITLE_ADD = 'Add a Student';
export const FORM_TITLE_EDIT = 'Edit Student Profile';
