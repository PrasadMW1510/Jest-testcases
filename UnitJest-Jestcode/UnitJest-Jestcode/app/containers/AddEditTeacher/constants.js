/*
 *
 * AddEditTeacher constants
 *
 */

export const FORM_TEACHER_PROFILE = 'form_teacher_profile';

export const ADD_EDIT_TEACHER_REQUEST = 'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST';
export const ADD_EDIT_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_FAILURE';

export const POST_ADD_TEACHER_REQUEST = 'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST';
export const POST_ADD_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST_SUCCESS';
export const POST_ADD_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST_FAILURE';

export const POST_SAVE_TEACHER_REQUEST = 'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST';
export const POST_SAVE_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST_SUCCESS';
export const POST_SAVE_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST_FAILURE';

export const POST_SAVE_TEACHER_MIA_REQUEST = 'app/AddEditTeacher/POST_SAVE_TEACHER_MIA_REQUEST';

export const ADD_EDIT_TEACHER_REQUEST_PROFILE_DETAILS_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PROFILE_DETAILS_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_PERMISSIONS_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PERMISSIONS_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_SCHOOL_AND_CLASSES_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_SCHOOL_AND_CLASSES_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_PASSWORD_CONFIG_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PASSWORD_CONFIG_SUCCESS';
