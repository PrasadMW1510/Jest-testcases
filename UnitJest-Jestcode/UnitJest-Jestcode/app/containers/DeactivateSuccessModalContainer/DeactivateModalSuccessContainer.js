/**
 *
 * DeactivateSuccessModalContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { USER_ORG } from 'containers/App/constants';
import { isUserTypeAdminOrTech } from 'utils/utilities';
import { makeSelectProfileUserType, makeSelectLoginUserOrg } from 'containers/App/selectors';
import { hideModal } from 'containers/ModalController/actions';
import { schoolRedirection } from 'containers/SmartBarContainer/actions';
import { schoolUserLoginFlowRequest } from 'containers/App/actions';
import { usageSummaryRequest } from 'containers/UsageSummaryContainer/actions';
import DeactivateTeacherSuccessModal from 'components/DeactivateTeacherSuccessModal/DeactivateTeacherSuccessModal';
import { makeSelectSchoolId } from 'containers/SmartBarContainer/selectors';

export function DeactivateModalSuccessContainer(props) {
  const handleYes = e => {
    e.preventDefault();
    props.hideModal();
    props.hideModal();
    props.schoolRedirection(props.activeSmartBarSchoolId);
    if (props.profileOrgType === USER_ORG.School && isUserTypeAdminOrTech(props.profileUserType)) {
      props.schoolUserLoginFlowRequest();
      props.usageSummaryRequest();
    }
  };

  return <DeactivateTeacherSuccessModal isOpen onYes={handleYes} />;
}

DeactivateModalSuccessContainer.propTypes = {
  hideModal: PropTypes.func.isRequired,
  schoolRedirection: PropTypes.func,
  schoolUserLoginFlowRequest: PropTypes.func,
  usageSummaryRequest: PropTypes.func,
  activeSmartBarSchoolId: PropTypes.string,
  profileOrgType: PropTypes.string,
  profileUserType: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  activeSmartBarSchoolId: makeSelectSchoolId(),
  profileOrgType: makeSelectLoginUserOrg(),
  profileUserType: makeSelectProfileUserType(),
});

const withConnect = connect(mapStateToProps, {
  hideModal,
  schoolRedirection,
  schoolUserLoginFlowRequest,
  usageSummaryRequest,
});

export default compose(withRouter, withConnect)(DeactivateModalSuccessContainer);
