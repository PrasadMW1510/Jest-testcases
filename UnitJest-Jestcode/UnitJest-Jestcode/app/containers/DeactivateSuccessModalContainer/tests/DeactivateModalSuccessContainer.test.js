import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { USER_ORG, USER_TYPE } from 'containers/App/constants';
import { DeactivateModalSuccessContainer } from '../DeactivateModalSuccessContainer';

describe('<DeactivateModalSuccessContainer />', () => {
  let wrapper = null;
  let schoolRedirection = null;
  let schoolUserLoginFlowRequest = null;
  let mockHideModal = null;
  let profileUserType = null;
  let profileOrgType = null;
  let usageSummaryRequest = null;
  let activeSmartBarSchoolId = null;
  const mockEvent = { preventDefault: jest.fn() };

  beforeEach(() => {
    mockHideModal = jest.fn();
    schoolRedirection = jest.fn();
    profileUserType = USER_TYPE.Tech;
    profileOrgType = USER_ORG.School;
    schoolUserLoginFlowRequest = jest.fn();
    usageSummaryRequest = jest.fn();
    activeSmartBarSchoolId = '123456';

    wrapper = shallow(
      <DeactivateModalSuccessContainer
        hideModal={mockHideModal}
        schoolRedirection={schoolRedirection}
        schoolUserLoginFlowRequest={schoolUserLoginFlowRequest}
        usageSummaryRequest={usageSummaryRequest}
        profileUserType={profileUserType}
        profileOrgType={profileOrgType}
        activeSmartBarSchoolId={activeSmartBarSchoolId}
      />
    );
  });

  it('should have a DeactivateTeacherSuccessModal', () => {
    const modal = wrapper.find('DeactivateTeacherSuccessModal');
    expect(modal).toBeDefined();
  });

  it('should deactivate modal close when user agrees', () => {
    const modal = wrapper.find('DeactivateTeacherSuccessModal');
    modal.prop('onYes')(mockEvent);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should deactivate modal close when user agrees', () => {
    const modal = wrapper.find('DeactivateTeacherSuccessModal');
    modal.prop('onYes')(mockEvent);
    profileUserType = USER_TYPE.Tech;
    profileOrgType = USER_ORG.School;
    wrapper.setProps({ profileUserType: USER_TYPE.Tech, profileOrgType: USER_ORG.School });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
