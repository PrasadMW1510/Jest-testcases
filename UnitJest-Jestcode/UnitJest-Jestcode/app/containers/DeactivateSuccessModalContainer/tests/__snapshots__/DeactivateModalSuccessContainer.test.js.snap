// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeactivateModalSuccessContainer /> should deactivate modal close when user agrees 1`] = `
<DeactivateTeacherSuccessModal
  isOpen={true}
  onYes={[Function]}
/>
`;

exports[`<DeactivateModalSuccessContainer /> should deactivate modal close when user agrees 2`] = `
<DeactivateTeacherSuccessModal
  isOpen={true}
  onYes={[Function]}
/>
`;
